### DISCUSSION TITLE
My understanding about how this brilliant prompt works

### DISCUSSION THREAD

**Discussion by darcula1993** [JushBJJ/Mr.-Ranedeer-AI-Tutor#64] [General]
1. This prompt define a new kind of programming language. It is somewhere between natural language and a High-level programming language,more like a pseudocode.
2. This programming language use chatgpt4 as compiler/interpreter to execute the code.
3. Code block start with `[BEGIN]` and end with `[END]`, just like C++ use {}.
4. It define `functions` (or maybe `class` ) use square brackets `[ ]`. Functions can take parameters as input,using `Args`.
5. `[INSTRUCTIONS]` under functions works like `__init__` in python class. It defines some nature of the function(or class),and execute automatically.
6. Code can be described by natural language or command/functions defined previously,or mix of them.

Good job bro! bravo!

**Comment by JushBJJ** [comment]
Correct

**Reply by uma-sathish** [reply]
can you please help me. how to I learn this?

**Comment by nnnpa31** [comment]
In fact, you know, all these points you analyzed are obvious for this amazing and great spell. The bigger point of this interpretation is that it proves as a PoC a more effective way to interact with GPT4, it proves the feasibility of "programming" on GPT4 and provides one of the ways to do so, and such an exploration is so meaningful that every future GPT4-related prompt may start from Every future GPT4-related prompt may benefit from this PoC.

I still can't believe that GPT4 as a natural language inference model has such a complete and accurate understanding of "overhead programming languages", which is amazing!

**Comment by darcula1993** [comment]
Yes. Most programming language is just a variation of English. If LLM can understand English and most of programming language, then "programming" is just another kind of "talking" with it.

**Comment by A-F-V** [comment]
Reminds me a little of a .toml file actually

**Comment by TYRONEMICHAEL** [comment]
I am still bewildered how it knows how to execute the code. Why wasn't it chosen to be written in one of the popular languages? And finally why was it updated from using JSON to using using its own programming language (more power/functionality
 perhaps)?

**Reply by TYRONEMICHAEL** [reply]
I am not sure if I'm being silly, but I tried writing some code in TS and asked it to execute it, but it failed. Why does this work?
